src/managers/PermissionOverwriteManager.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const BaseManager = require('./BaseManager');
+const { TypeError } = require('../errors');
+const PermissionOverwrites = require('../structures/PermissionOverwrites');
+const Role = require('../structures/Role');
+const Collection = require('../util/Collection');
+const { OverwriteTypes } = require('../util/Constants');
+
+/**
+ * Manages API methods for guild channel permission overwrites and stores their cache.
+ * @extends {BaseManager}
+ */
+class PermissionOverwriteManager extends BaseManager {
+  constructor(channel, iterable) {
+    super(channel.client, iterable, PermissionOverwrites);
+
+    /**
+     * The channel of the permission overwrite this manager belongs to
+     * @type {GuildChannel}
+     */
+    this.channel = channel;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, PermissionOverwrites>}
+   * @name PermissionOverwriteManager#cache
+   */
+
+  add(data, cache) {
+    return super.add(data, cache, { extras: [this.channel] });
+  }
+
+  /**
+   * Replaces the permission overwrites in this channel.
+   * @param {OverwriteResolvable[]|Collection<Snowflake, OverwriteResolvable>} overwrites
+   * Permission overwrites the channel gets updated with
+   * @param {string} [reason] Reason for updating the channel overwrites
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * message.channel.permissionOverwrites.set([
+   *   {
+   *      id: message.author.id,
+   *      deny: [Permissions.FLAGS.VIEW_CHANNEL],
+   *   },
+   * ], 'Needed to change permissions');
+   */
+  async set(overwrites, reason) {
```

Comment:
Personally, I'd be more of a fan of calling this `overwrite` than `set`, since we're replacing everything. CC @iCrawl @kyranet tho

Reply:
Discussed in the Discord server and it was rather more consistent with `set` (e.g. `GuildMember#roles#set`, etc.)